### PR TITLE
(CFACT-227) Move logging and boost-nowide into leatherman 

### PR DESCRIPTION
**Pull request by branan**
I also pulled basic CFLAG setting out of cfacter, since leatherman
needs the logic to build itself as C++11. This means cfacter now
relies on leatherman to set up its core cflags as well (to ensure we
don't screw up our ABI between leatherman libraries and the apps that
use it)

### CODE REVIEW

File: logging/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+find_package(Boost 1.54 REQUIRED COMPONENTS log)
+
+include_directories(${Boost_INCLUDE_DIRS})
+include_directories(${LEATHERMAN_CATCH_INCLUDE})
+include_directories(${LEATHERMAN_NOWIDE_INCLUDE})
+include_directories(${LEATHERMAN_LOCALE_INCLUDE})
+include_directories(${LEATHERMAN_LOGGING_INCLUDE})
+
+add_definitions("-DLEATHERMAN_LOGGING_NAMESPACE=\"leatherman.logging\"")
```
Comment (branan) [inline logging/CMakeLists.txt]: This is gross, and needs to be done in each place that has a different logging context (my cfacter changes have it added in 3 places). I can definitely make a `leatherman_logging_context` macro, but that doesn't change the facter that each `.cc` file needs to know its own logging context somehow.

File: logging/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+find_package(Boost 1.54 REQUIRED COMPONENTS log)
+
+include_directories(${Boost_INCLUDE_DIRS})
+include_directories(${LEATHERMAN_CATCH_INCLUDE})
```
Comment (branan) [inline logging/CMakeLists.txt]: I'm going to wrap these includes in a macro that ensure the dependency libraries are enabled, and bail if not.

When we're in the `find_package` world, I'll be sure to enable proper dependency tracking between leatherman components.

**Comment by ploubser** [comment]
We discussed being able to add line number and thread id to the log lines. This still on the cards?

**Comment by branan** [comment]
Gross, this didn't do a great job with the rename of  `leatherman.cmake -> internal.cmake` and adding a new `leatherman.cmake`. Makes the diff hard to see there. I'm going to try splitting that up into multiple commits so git can track the rename.

**Comment by branan** [comment]
Nope :(

**Comment by branan** [comment]
Please look at comit 73d9444 directly to understand the changes to `internal.cmake` and the addition of the new API-level `leatherman.cmake`. The github UI is failing us here.

**Comment by MikaelSmith** [comment]
I'd like to see Travis and AppVeyor applied on this PR.

**Comment by MikaelSmith** [comment]
I'd like to see the Readme talk about configuring logging.

**Comment by branan** [comment]
We're back to the vendored version of boost-nowide. Travis is in #3, appveyor is gonna take a bit of learning before I can just toss it in.

README update incoming.

**Comment by branan** [comment]
Rebased on top of the version with a travis config

**Comment by branan** [comment]
Amazingly the travis file that worked on my laptop is failing when actually run in travis >_<

**Comment by branan** [comment]
WOOHOO, green travis run!

There are still likely to be a few changes for the appveyor failure I'm seeing in puppetlabs/cfacter#317

**Comment by branan** [comment]
puppetlabs/cfacter#317 passes in appveyor with the latest version of this code.

**Comment by kylog** [comment]
Okay one minor typo in a comment, and I'm :+1:

**Comment by branan** [comment]
* Fixed a couple of types
* Added docs for configuring logging

**Pull request merged**